channeld/channeld.c

```diff
@@ -1964,10 +1967,28 @@ static void send_revocation(struct peer *peer,
 
 	peer->splice_state->await_commitment_succcess = false;
 
+	/* Now that the master has persisted the new commitment advance the HSMD
+	 * and fetch the revocation secret for the old one. */
+	struct secret old_secret2;
+	struct pubkey next_point2;
+	if (!hsm_is_capable(peer->hsm_capabilities, WIRE_HSMD_REVOKE_COMMITMENT_TX)) {
+		/* Prior to HSM_VERSION 5 we use the old_secret
+		 * received earlier from validate_commitment_tx. */
+		memcpy(&old_secret2, old_secret, sizeof(old_secret2));
+		memcpy(&next_point2, next_point, sizeof(next_point2));
```

Comment:
I always prefer assignment to memcpy, since it's typesafe (and compiler will do the same thing).  `old_secret2 = *old_secret;` and `next_point2 = *next_pointl`.